pkg/file/splitter/internal/job.go

```diff
@@ -0,0 +1,225 @@
+	hasher     bmt.Hash // underlying hasher used for hashing the tree
+	buffer     []byte   // keeps data and hashes, indexed by cursors
+	logger     logging.Logger
+}
+
+// NewSimpleSplitterJob creates a new SimpleSplitterJob.
+//
+// The spanLength is the length of the data that will be written.
+func NewSimpleSplitterJob(ctx context.Context, store storage.Storer, spanLength int64, logger logging.Logger) *SimpleSplitterJob {
+	p := bmtlegacy.NewTreePool(hashFunc, swarm.Branches, bmtlegacy.PoolSize)
+	return &SimpleSplitterJob{
+		ctx:        ctx,
+		store:      store,
+		spanLength: spanLength,
+		sumCounts:  make([]int, levelBufferLimit),
+		cursors:    make([]int, levelBufferLimit),
+		hasher:     bmtlegacy.New(p),
+		buffer:     make([]byte, swarm.ChunkSize*levelBufferLimit),
+		logger:     logger,
+	}
+}
+
+// Write adds data to the file splitter.
+func (j *SimpleSplitterJob) Write(b []byte) (int, error) {
+	if len(b) > swarm.ChunkSize {
+		return 0, fmt.Errorf("Write must be called with a maximum of %d bytes", swarm.ChunkSize)
+	}
+	j.length += int64(len(b))
+	if j.length > j.spanLength {
+		return 0, errors.New("Write past span length")
+	}
+
+	err := j.writeToLevel(0, b)
+	if err != nil {
+		return 0, err
+	}
+	if j.length == j.spanLength {
+		err := j.hashUnfinished()
+		if err != nil {
+			return 0, file.NewHashError(err)
+		}
+		err = j.moveDanglingChunk()
+		if err != nil {
+			return 0, file.NewHashError(err)
+		}
+
+	}
+	return len(b), nil
+}
+
+// Sum returns the Swarm hash of the data.
+func (j *SimpleSplitterJob) Sum(b []byte) []byte {
+	return j.digest()
+}
+
+// writeToLevel writes to the data buffer on the specified level.
+// It calls sum if chunk boundary is reached and recursively calls this function for
+// the next level with the acquired bmt hash
+//
+// It adjusts the relevant levels' cursors accordingly.
+func (s *SimpleSplitterJob) writeToLevel(lvl int, data []byte) error {
+	copy(s.buffer[s.cursors[lvl]:s.cursors[lvl]+len(data)], data)
+	s.cursors[lvl] += len(data)
+	if s.cursors[lvl]-s.cursors[lvl+1] == swarm.ChunkSize {
+		ref, err := s.sumLevel(lvl)
+		if err != nil {
+			return err
+		}
+		err = s.writeToLevel(lvl+1, ref)
+		if err != nil {
+			return err
+		}
+		s.cursors[lvl] = s.cursors[lvl+1]
+	}
+	return nil
+}
+
+// sumLevel calculates and returns the bmt sum of the last written data on the level.
+//
+// TODO: error handling on store write fail
+func (s *SimpleSplitterJob) sumLevel(lvl int) ([]byte, error) {
+	s.sumCounts[lvl]++
+	spanSize := file.Spans[lvl] * swarm.ChunkSize
+	span := (s.length-1)%spanSize + 1
+
+	sizeToSum := s.cursors[lvl] - s.cursors[lvl+1]
+
+	s.hasher.Reset()
+	err := s.hasher.SetSpan(span)
+	if err != nil {
+		return nil, err
+	}
+	_, err = s.hasher.Write(s.buffer[s.cursors[lvl+1] : s.cursors[lvl+1]+sizeToSum])
+	if err != nil {
+		return nil, err
+	}
+	ref := s.hasher.Sum(nil)
+	addr := swarm.NewAddress(ref)
+	ch := swarm.NewChunk(addr, s.buffer[s.cursors[lvl+1]:s.cursors[lvl]])
```

Comment:
so you actually don't need to have the span in the chunk data here before the actual data?

Reply:
`SetSpan` takes care of that, see above.

Reply:
yeah that's in regards to the hasher, that i have seen, but what i'm asking is, because the span is not encoded into the chunk data, once that chunk gets reloaded from the chunk store and passes through the validator will it still pass the content addressed validation?

Reply:
that would be a good test case